### DISCUSSION TITLE
Database Support

### DISCUSSION THREAD

**Discussion by pra-cloud** [Unleash/unleash#2921] [Q&A] [answered]
Does unleash support any other database instead of the PostgreSQL ??

**Comment by pransh15** [comment]
Not officially, there are workarounds but I wouldn't recommend that. What's the usecase here @pra-cloud ? 😄

**Reply by thostr** [reply]
Hi. I have a use case where our customer relies heavily on Oracle RDBMS, and supporting PostgreSQL is currently not an option.

@pransh15  you say other databaseses are not officially supported but there are workarounds. What's needed to get database support for other database engines say Oracle RDBMS 12+? Is there any interest / does the current architecture allow for multiple database providers? I could be interested in spending time getting support in place for this, if it's something that will be a candidate for official support at some time (or it's possible to use with official versions of Unleash at least).

**Reply by sjaanus** [reply]
Hey there @thostr ! 😊

Thank you for reaching out and expressing interest in expanding database support for Unleash. We totally understand the importance of Oracle RDBMS, especially in certain enterprise scenarios.

However, as of now, Unleash is specifically tailored for PostgreSQL. This is primarily because we have a substantial number of migration files, SQL syntaxes, and specific optimizations that have been built for PostgreSQL over time. Introducing support for another database, like Oracle RDBMS, would require extensive rework and might introduce unforeseen challenges and complexities.

Thank you for your understanding and continuous support! Let us know if you have any other questions or ideas.

**Reply by sighphyre** [reply]
@thostr Just to give some context for @pransh15's comment, we've seen some users attempt to use cockroach db as a replacement for Postgres. That's not really possible with other databases

**Comment by ykostsik** [comment]
Hi all,
What PostgreSQL version is required/recommended to use with Unleash?
I have two dockerized deployments, and one of them works successfully with 14.6.
But the second one cannot start using postgres 9.2.
Unleash v.4.19.1

**Reply by ivarconr** [reply, marked as answer]
Unleash requires at least PostgreSQL  v10 or later (also [stated in the docs](https://docs.getunleash.io/reference/deploy/getting-started))

PostgreSQL v14 is preffered if you can, due to a few performance optimization that also Unleash benefits from.
